### DISCUSSION TITLE
Do properties need to be defined in the settings class for each setting?

### DISCUSSION THREAD

**Discussion by rich06** [spatie/laravel-settings#226] [Q&A]
I'm a little confused about settings management and the associated properties defined in the settings class.

Does a property have to be defined in the settings class for it to be managed by this package? Given the example used in the docs can I create a new 'general' setting using a web form without a property for the setting being defined in the class first? So if I wanted to create a general setting called 'site_tag' I would have to define this as a property in the class as follows:-

`    public string $site_tag;
` 

Is that the case?

Also same principal goes for delete - can a setting be deleted if it has a property defined in the class or does it have to be removed from the class as well?

**Comment by rubenvanassche** [comment]
Migrations normally work outside of the settings class scope, so they can work on settings without them being in the class.

If you want to write settings in settings classes then properties should always be defined.
